src/Microsoft.AspNetCore.Server.Testing/Deployers/RemoteDeployer/RemoteDeployer.cs

```diff
@@ -0,0 +1,279 @@
+            {
+                throw new ArgumentException($"Invalid value '{_deploymentParameters.ServerName}' for {nameof(RemoteDeploymentParameters.ServerName)}");
+            }
+
+            if (string.IsNullOrWhiteSpace(_deploymentParameters.ServerAccountName))
+            {
+                throw new ArgumentException($"Invalid value '{_deploymentParameters.ServerAccountName}' for {nameof(RemoteDeploymentParameters.ServerAccountName)}." +
+                    " Account credentials are required to enable creating a powershell session to the remote server.");
+            }
+
+            if (string.IsNullOrWhiteSpace(_deploymentParameters.ServerAccountPassword))
+            {
+                throw new ArgumentException($"Invalid value '{_deploymentParameters.ServerAccountPassword}' for {nameof(RemoteDeploymentParameters.ServerAccountPassword)}." +
+                    " Account credentials are required to enable creating a powershell session to the remote server.");
+            }
+
+            if (string.IsNullOrWhiteSpace(_deploymentParameters.RemoteServerFileShare))
+            {
+                throw new ArgumentException($"Invalid value for {nameof(RemoteDeploymentParameters.RemoteServerFileShare)}." +
+                    " . A file share is required to copy the application's published output.");
+            }
+
+            if (string.IsNullOrWhiteSpace(_deploymentParameters.RemoteServerRelativeExecutablePath))
+            {
+                throw new ArgumentException($"Invalid value for {nameof(RemoteDeploymentParameters.RemoteServerRelativeExecutablePath)}." +
+                    " This is the name of the executable in the published output which needs to be executed on the remote server.");
+            }
+        }
+
+        public override DeploymentResult Deploy()
+        {
+            if (_isDisposed)
+            {
+                throw new ObjectDisposedException("This instance of deployer has already been disposed.");
+            }
+
+            // Publish the app to a local temp folder on the machine where the test is running
+            DotnetPublish();
+
+            var folderId = Guid.NewGuid().ToString();
+            _deployedFolderPathInFileShare = Path.Combine(_deploymentParameters.RemoteServerFileShare, folderId);
+
+            DirectoryCopy(
+                _deploymentParameters.PublishedApplicationRootPath,
+                _deployedFolderPathInFileShare,
+                copySubDirs: true);
+            Logger.LogInformation($"Copied the locally published folder to the file share path '{_deployedFolderPathInFileShare}'");
+
+            RunScript("StartServer");
+
+            string uri;
+            if (_deploymentParameters.ServerType == ServerType.IIS)
+            {
+                uri = $"http://{_deploymentParameters.ServerName}:80/";
+            }
+            else
+            {
+                uri = $"http://{_deploymentParameters.ServerName}:5000/";
```

Comment:
You'll have to make sure none of the tests execute concurrently or they'll run into resource conflicts.